core/lib/libtomcrypt/src/headers/tomcrypt_private.h

```diff
@@ -177,6 +177,11 @@ void ocb3_int_xor_blocks(unsigned char *out, const unsigned char *block_a, const
 #define mp_toradix(a, b, c)          ltc_mp.write_radix(a, b, c)
 #define mp_unsigned_bin_size(a)      ltc_mp.unsigned_size(a)
 #define mp_to_unsigned_bin(a, b)     ltc_mp.unsigned_write(a, b)
+#define mp_to_unsigned_bin2(a, b, c) \
+        do { \
+                void *_a = (a); \
+                mp_to_unsigned_bin(_a, (b) + (c) - mp_unsigned_bin_size(_a)); \
```

Comment:
I assume the `- mp_unsigned_bin_size(_a)` is to adjust it to start at the offset `b`.
This macro is a bit special purpose, perhaps it should rather go into `acipher_helpers.h`.

Reply:
> I assume the `- mp_unsigned_bin_size(_a)` is to adjust it to start at the offset `b`.

Yes, well, to *not* start at `b` actually. `mp_to_unsigned_bin()` would write the MSB of `a` at `b`. We want to leave zeros at the start of the buffer instead. Call that "right justify" if you wish, a bit like `printf("08x", n)` for instance...

> This macro is a bit special purpose, perhaps it should rather go into `acipher_helpers.h`.

It is useful each time one works with fixed-size integer buffers. Apparently this is not so frequent or LTC would have introduced this macro already. I will move it to `acipher_helpers.h`.